### PR TITLE
Stormwind: Fix "The Missing Diplomat - Part 8" quest

### PR DESCRIPTION
**Pull request by miraco**
Update the RP between Dashel stonefist and old town thug

Based on sniff data and ClassicPTR tests.

### Proof

- None

### Issues

- None

### How2Test

- None

### Todo / Checklist

- [X] None

### CODE REVIEW

File: src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp [hunk 1]
```diff
@@ -143,51 +231,218 @@ struct npc_dashel_stonefistAI : public CombatAI
...
         }
     }
 
     void HandleEndEvent()
     {
-        SetDeathPrevention(false);
-        DoScriptText(SAY_STONEFIST_3, m_creature);
-        m_creature->CastSpell(nullptr, SPELL_UNKILLABLE_OFF, TRIGGERED_OLD_TRIGGERED);
+        // Occurs only if thugs are alive
+        if(m_thugsAlive)
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp:276]: Missing space :D

File: src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp [hunk 2]
```diff
@@ -113,28 +136,93 @@ struct npc_dashel_stonefistAI : public CombatAI
...
-        ResetTimer(DASHEL_START_EVENT, 3000);
+        ResetTimer(DASHEL_START_EVENT, 2000);
+        m_questFightStarted = true;
+    }
+
+    void HandleStartEvadeEvent()
+    {
+        // Move TargetHome after Emote
+        m_creature->GetMotionMaster()->MoveTargetedHome();
+        DisableTimer(DASHEL_EVADE_EVENT);
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp:208]: This isnt necessary. Custom timers dont repeat unless you explicitly tell them to.

File: src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp [hunk 3]
```diff
@@ -143,51 +231,218 @@ struct npc_dashel_stonefistAI : public CombatAI
...
+                    break;
+                }
+                case 5:
+                {
+                    // Set quest completed
+                    if (Player* player = m_creature->GetMap()->GetPlayer(m_playerGuid))
+                        player->AreaExploredOrEventHappens(QUEST_MISSING_DIPLO_PT8);
+
+                    Reset();
+                    DisableTimer(DASHEL_END_EVENT);
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp:319]: Again no need for this.

File: src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp [hunk 4]
```diff
@@ -143,51 +231,218 @@ struct npc_dashel_stonefistAI : public CombatAI
...
-        m_creature->CastSpell(nullptr, SPELL_UNKILLABLE_OFF, TRIGGERED_OLD_TRIGGERED);
+        // Occurs only if thugs are alive
+        if(m_thugsAlive)
+        {
+            uint32 timer = 0;
+            switch (m_phaseThugEventStage)
+            {
+                case 1:
+                {
+                    if (Creature* pThug = m_creature->GetMap()->GetCreature(m_thugs[0]))
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp:283]: name them just thug

File: src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp [hunk 5]
```diff
@@ -113,28 +136,93 @@ struct npc_dashel_stonefistAI : public CombatAI
     {
         AddTimerlessCombatAction(DASHEL_LOW_HP, true);
         AddCustomAction(DASHEL_START_EVENT, true, [&]() { HandleStartEvent(); });
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp:138]: Prolly also want to mark this one if its combat/ooc/any

File: src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp (hunk 4 above)
Comment (miraco) [reply src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp:283]: You mean to change pThug to thug or m_thugs to m_thug?

Comment (killerwife) [reply src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp:282]: pThug to thug

Comment (miraco) [reply src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/stormwind_city.cpp]: should be done now

**Pull request merged**